### ISSUE TITLE
I'm wondering if we need this function at all (validation.rs::credential)

### ISSUE THREAD

**Issue by franziskuskiefer** [openmls/openmls#754]
I'm wondering if we need this function at all after #717. Initially, this served the purpose to make it easy for an app to know who the sender of the message was by providing the credential instead of just the leaf index.
This might be out-of-scope for this PR because it would require more refactoring, but I wanted to bring it up here, given how much code is changed here. I'm fine merging as-is.

_Originally posted by @raphaelrobert in https://github.com/openmls/openmls/pull/746#discussion_r796828856_

**Comment by keks** [comment]
Is this still relevant? I just recently made changes to that function because it was wrong. Also, it looks like issue #717 referenced above was about not using leaf node indices - but I think we still use a lot more leaf node indices than key package refs? I have a feeling this might be leftovers from an old refactor that is just not relevant anymore, but let me know if I'm wrong.
